zutil.h

```diff
@@ -137,15 +137,11 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #  endif
 #endif
 
-#if defined(MACOS) || defined(TARGET_OS_MAC)
+#if TARGET_OS_OSX
 #  define OS_CODE  7
 #  ifndef Z_SOLO
 #    if defined(__MWERKS__) && __dest_os != __be_os && __dest_os != __win32_os
```

Comment:
The history of this code is interesting.
zlib 0.71 (the first version in git) had:
```
#ifdef MACOS
#  define OS_CODE  0x07
#endif
```

zlib 1.0.8 added the fdopen thing, but differently:
```
#if defined(MACOS) || defined(TARGET_OS_MAC)
#  define OS_CODE  0x07
#  ifndef fdopen
#    define fdopen(fd,mode) NULL /* No fdopen() */
#  endif
#endif
#if defined(__MWERKS__) && !defined(fdopen)
#  if __dest_os != __be_os && __dest_os != __win32_os
#    define fdopen(fd,mode) NULL
#  endif
#endif
```
The corresponding changelog:
> - check for TARGET_OS_MAC in addition to MACOS (Brad Pettit)
> - do not use fdopen for Metrowerks on Mac (Brad Pettit))

zlib 1.1.3 changed it to:
```
#if defined(MACOS) || defined(TARGET_OS_MAC)
#  if defined(__MWERKS__) && __dest_os != __be_os && __dest_os != __win32_os
#    include <unix.h> /* for fdopen */
#  else
#    ifndef fdopen
#      define fdopen(fd,mode) NULL /* No fdopen() */
#    endif
#  endif
#endif
```
I think the corresponding changelog entry is:
> - Support gzdopen on Mac with Metrowerks (Jason Linhart)

Presumably, Metrowerks was defining TARGET_OS_MAC, and probably didn't define TARGET_OS_OSX, so my assumption here is that this change breaks Metrowerks. However, it's also a long gone compiler, so maybe this part can go entirely, which leaves us with two different OS_CODE values for macOS, one of which was added in ce12c5cd00628bf8f680c98123a369974d32df15